module/BlockBingoData.cpp

```diff
@@ -39,34 +62,204 @@ bool BlockBingoData::initCrossCircle(int x, int y, CrossCircle crossCircle)
+          blockCircleNumber++;
+        }
+        setBlockCircle(i, j, initBlockCircle);
+      }
+    }
+  }
+
+  // 数字カードの初期化
+  initCardNumber(controller.getCourseInfo(ETROBOC_COURSE_INFO_CARD_NUMBER));
+
+  // ブロックの初期位置を設定する
+  initBlock(Color::black, controller.getCourseInfo(ETROBOC_COURSE_INFO_BLOCK_POS_BLACK1));
+  initBlock(Color::black, controller.getCourseInfo(ETROBOC_COURSE_INFO_BLOCK_POS_BLACK2));
+  initBlock(Color::red, controller.getCourseInfo(ETROBOC_COURSE_INFO_BLOCK_POS_RED1));
+  initBlock(Color::red, controller.getCourseInfo(ETROBOC_COURSE_INFO_BLOCK_POS_RED2));
+  initBlock(Color::yellow, controller.getCourseInfo(ETROBOC_COURSE_INFO_BLOCK_POS_YELLOW1));
+  initBlock(Color::yellow, controller.getCourseInfo(ETROBOC_COURSE_INFO_BLOCK_POS_YELLOW2));
+  initBlock(Color::blue, controller.getCourseInfo(ETROBOC_COURSE_INFO_BLOCK_POS_BLUE1));
+  initBlock(Color::blue, controller.getCourseInfo(ETROBOC_COURSE_INFO_BLOCK_POS_BLUE2));
+  initBlock(Color::green, controller.getCourseInfo(ETROBOC_COURSE_INFO_BLOCK_POS_GREEN1));
+  initBlock(Color::green, controller.getCourseInfo(ETROBOC_COURSE_INFO_BLOCK_POS_GREEN2));
+}
+
+void BlockBingoData::initCardNumber(int cardNumber_)
 {
   cardNumber = cardNumber_;
 }
 
-BlockCircle BlockBingoData::getBlockCircle(int x, int y)
+void BlockBingoData::initBlock(Color initColor, int coordinate)
 {
-  return blockCircleCoordinate[x][y];
+  int x, y;
+  BlockCircle blockCircle;
+  CrossCircle crossCircle;
+
+  switch(coordinate) {
+      // ブロックサークル
+    case 1:
+      x = 1, y = 1;
+      break;
+    case 2:
+      x = 3, y = 1;
+      break;
+    case 3:
+      x = 5, y = 1;
+      break;
+    case 4:
+      x = 0, y = 3;
+      break;
+    case 5:
+      x = 5, y = 3;
+      break;
+    case 6:
+      x = 0, y = 5;
+      break;
+    case 7:
+      x = 3, y = 5;
+      break;
+    case 8:
+      x = 5, y = 5;
+      break;
+
+      // 交点サークル
+    case 65:
+      x = 0, y = 0;
+      break;
+    case 67:
+      x = 4, y = 0;
+      break;
+    case 70:
+      x = 2, y = 2;
+      break;
+    case 72:
+      x = 6, y = 2;
+      break;
+    case 74:
+      x = 0, y = 4;
+      break;
+    case 76:
+      x = 4, y = 4;
+      break;
+    case 81:
+      x = 2, y = 6;
+      break;
+    case 83:
+      x = 6, y = 6;
+      break;
+    default:
+      x = -1, y = -1;
+  }
+
+  if((coordinate > 0) && (coordinate <= 8)) {
+    blockCircle = getBlockCircle(x, y);
+    blockCircle.block.blockColor = initColor;
+    setBlockCircle(x, y, blockCircle);
+  } else {
+    crossCircle = getCrossCircle(x, y);
+    crossCircle.block.blockColor = initColor;
+    setCrossCircle(x, y, crossCircle);
+  }
 }
 
-CrossCircle BlockBingoData::getCrossCircle(int x, int y)
+Color BlockBingoData::getBlockCircleColor(int circleNumber)
 {
-  return crossCircleCoordinate[x][y];
+  // ブロックサークルにない数字の時のエラー処理
+  if((circleNumber < 1) && (circleNumber > 8)) {
```

Comment:
「1未満、または、8より大きいとき」だよね？
たぶん、getBlockCircleColor関数の異常時のテストが書かれていないと思う
```suggestion
  if((circleNumber < 1) || (circleNumber > 8)) {
```